libsolidity/interface/CompilerStack.h

```diff
@@ -117,43 +117,39 @@ class CompilerStack: boost::noncopyable
 	static boost::optional<Remapping> parseRemapping(std::string const& _remapping);
 
 	/// Sets path remappings.
+	/// Must be set before parsing.
 	void setRemappings(std::vector<Remapping> const& _remappings);
 
 	/// Sets library addresses. Addresses are cleared iff @a _libraries is missing.
-	/// Will not take effect before running compile.
-	void setLibraries(std::map<std::string, h160> const& _libraries = std::map<std::string, h160>{})
-	{
-		m_libraries = _libraries;
-	}
+	/// Must be set before parsing.
+	void setLibraries(std::map<std::string, h160> const& _libraries = std::map<std::string, h160>{});
 
 	/// Changes the optimiser settings.
-	/// Will not take effect before running compile.
-	void setOptimiserSettings(bool _optimize, unsigned _runs = 200)
-	{
-		m_optimize = _optimize;
-		m_optimizeRuns = _runs;
-	}
+	/// Must be set before parsing.
+	void setOptimiserSettings(bool _optimize, unsigned _runs = 200);
```

Comment:
Oh come on! What about reviewing my conflicting pull request first? :)

Reply:
DAMN! I wanted yours to be merged first as this is simpler to rebase.